grakn-core/src/main/java/ai/grakn/GraknComputer.java

```diff
@@ -50,19 +52,45 @@
     /**
      * Execute the given vertex program using a graph computer.
      *
+     * @param includesShortcut whether the graph computer should include shortcut edges
+     * @param types            instance types in the subgraph
+     * @param program          the vertex program
+     * @param mapReduce        a list of mapReduce job
+     * @return the result of the computation
+     * @see ComputerResult
+     */
+    @CheckReturnValue
+    ComputerResult compute(Boolean includesShortcut, Set<LabelId> types, VertexProgram program, MapReduce... mapReduce);
```

Comment:
All these overloaded methods are confusing, especially since the optional arguments are at the beginning, not the end. We can't move them to the end because of the varargs param though...

I've got two bad ideas: one is to make the `MapReduce` argument a `List`, so we can move the optional args to the end. Or we can make this a builder? I think autovalue can help here too.

Then you can write:
```
computer.builder(program).withShortcuts(true).types(types).mapReduce(reduce).build();
```

But this might be overkill. Any other ideas?